### ISSUE TITLE
Unobtainable odds and ends in Garden of Glass

### ISSUE THREAD

**Issue by TyreForHyre** [Vazkii/Botania#3101]
Version Information
Forge version: 28.1.99 
Botania version:  r1.11-373

# Further Information
Just a list of things that are unobtainable in a Garden of Glass world (or relying solely on a wandering trader/villager)

Plants: 
Bamboo (no current access to pandas, or jungle biomes for fishing loot)
Sweet berries
Kelp (available only through wandering trader)
Lily of the Valley (available only through wandering trader, no mana infusion recipe)
Cornflower (actually does appear through bonemealing grass due to it being a plains biome, no mana infusion recipe)
Sea Pickle (available only through wandering trader)

Other misc items: 
Coral blocks (available only through wandering trader)
Coral and coral fans (no access for bonemealing as no Warm Ocean biomes)
Sponge (no access to ocean monuments or elder guardians)
Heart of the Sea (no access buried treasure, also blocks off crafting of Conduits and Ring of Chordata)
Bucket of Cod (no normal, cold, lukewarm oceans access for spawning the mob, Fisherman sells them individually)
Bucket of Salmon (no lukewarm, cold, normal, frozen oceans or rives for spawning the mob)
Bucket of Pufferfish (no lukewarm, warm, and deep lukewarm oceans for spawning the mob, available through Wandering trader)
Bucket of Tropical Fish (no lukewarm, warm oceans for spawning mob, available through Wandering trader)
Ink sacs (actually available through fishing but the squid itself is unobtainable as no access to rivers, beach variants and oceanic biomes)
Bell (available only through Armorer)

**Comment by ToMe25** [comment]
I actually think the Bell and maybe the Ink sac are okay that way, but i am not sure about the Ink sac.

**Comment by williewillus** [comment]
a lot of these stem from the fact that gog worlds don't have normal biome layers. this should be fairly easy to fix

**Comment by williewillus** [comment]
with the gog real biome change and the mana infusion update most of these are fixed. some of the wandering-trader-only ones I left as-is (kelp, coral).

**Comment by auxiliary-character** [comment]
Maybe non-blaze mob spawners, too?

**Comment by williewillus** [comment]
nah, mob grinders should be used in skyblock (and the rates are good since you can build farms low and don't have to dig a perimeter)

**Comment by natedogith1** [comment]
It'd be nice to have the wandering-trader-only ones available from another source.  Maths suggests that if you're looking for a specific item (that's not one of the sixth trades), you only have about a 50% change of finding it after 8 spawns (which, at an an average of 3.5 hours per spawn, equals about 28 hours) (27 spawns will give you about 90% chance and 94.5 hours)

**Comment by luxor-g** [comment]
Amethysts are totally unobtainable in GoG.
Maybe add a recipe for Budding Amethysts?

**Comment by TheRealWormbo** [comment]
Structure-sensitive Loonium (PR #4639) will make a bunch of things available without trading, if there is a way to locate structures, even with the proposed void nether:

- all smithing templates for netherite upgrades and armor trims
- all pottery sherds
- chest loot music discs (it already provided cat and 13, but now also pigstep, otherside, relic, and 5)
- sculk catalyst (and thus all other sculk blocks), disk fragments, echo shards, and amethyst shards from ancient cities
- moss and bamboo from shipwrecks (also potatoes and carrots, so you don't need to hope to get them from zombies)
- sponge from ocean monuments
- heart of the sea from buried treasure (good luck locating a 1x1x1 block structure though, the Loonium does not generate structure maps, as it would be too laggy)
- netherite-related items (if we end up removing ancient debris from the Orechid Ignem) and gilded blackstone from bastion remnants
- tall grass, short/tall fern, cactus, oak/acacia/spruce saplings, sweet berries, and various crop seeds, as well as dead bushes (it might be faster to let Hydroangeas decay, though) from various villages
- bells from ruined portals

The question is, how do you locate such structures? The three obvious vanilla-like approaches would be a special map, a special compass, or a variant of the ender eye, but a player would need to somehow obtain them without requiring an item from them. Maybe there could be both a map and a compass/eye version for location, with maps being sold by cartographers.
Amethyst shards obtained from ancient city loot could probably get a budding amethyst recipe somehow (maybe a somewhat expensive alchemy catalyst upgrade from amethyst blocks, or a runic altar recipe, once we made the reagent for that not fixed to livingrock).

**Comment by QuickBASIC** [comment]
I've been playing a lot of Garden of Glass and I'd be a fan of a new monocle that shows structure borders but otherwise you have to explore to find. There are plenty of other mods that help find structures if the player wanted (for example the Structure Compass by Mrbysco) , but any maps or compasses in base Botania would feel a bit "cheaty". 

Make it late game materials and cost a lot of Mana per second to wear so you have to invest in finding structures but wouldn't wear it all the time.

Alternatively, you could do something like the Immersive Engineering survey tools in rod form á la dousing (Scrying Rod?) where it gives you a direction and rough distance in the chat when you right click but only if you're in a certain radius.

**Comment by TheRealWormbo** [comment]
> I'd be a fan of a new monocle that shows structure borders

That is an interesting idea, but may require some work, since structure bounding boxes (unlike e.g. biome borders) are not actually known to the client.

> […] any maps or compasses in base Botania would feel a bit "cheaty".

The thing is, if we rely on the Loonium for relevant materials, we need to provide a way to find the corresponding structures in base Botania, or at least GoG. Maybe access to that method is easier in GoG, but requires end-game materials in regular Botania.

**Comment by QuickBASIC** [comment]
I wonder if it might be easier to do something Eye of Ender like where a craftable item only gives a general direction for the nearest structure and some item for identifying it when you arrive.

Vanilla Minecraft already offers a way to find Woodland Mansions, Swamp Huts, Jungle Pyramids, Trial Chambers, and Ocean Monuments via Explorer maps sold by the Cartographer. Many mods already also implement villager trades for special maps. I wonder if a solution would be for Botania to offer a way to generate/create maps to other types of structures (thematically Cartographer trades wouldn't fit, but maybe via dropping them as loot in some way from mobs from the Loonium creating a gameplay loop where you setup a Loonium to get loot _and_ also maps to other types of structures.

**Comment by TheRealWormbo** [comment]
Yes, there have been various suggestions like that on the forum, such as a "lucky coin" that is tossed up and lands towards the general direction of the corresponding structure.
